source/shared/cpp/ObjectModel/BaseActionElement.h

```diff
@@ -40,6 +40,11 @@ namespace AdaptiveSharedNamespace
 
         ActionType GetElementType() const;
 
+        Mode GetElementMode() const;
+
+        void SetMode(Mode&& value);
+        void SetMode(const Mode& value);
```

Comment:
Enum is always used as primitive values (you may check out the other usage in the repo to find it). It's not object so l- & r- references are not necessary. Just
```
  Mode GetMode() const;
  void SetMode(const Mode mode);
```

Reply:
Done